base/reduce.jl

```diff
@@ -280,29 +280,65 @@ mapreduce_impl(f, op, A::AbstractArrayOrBroadcasted, ifirst::Integer, ilast::Int
 """
     mapreduce(f, op, itrs...; [init])
 
-Apply function `f` to each element(s) in `itrs`, and then reduce the result using the binary
-function `op`. If provided, `init` must be a neutral element for `op` that will be returned
-for empty collections. It is unspecified whether `init` is used for non-empty collections.
-In general, it will be necessary to provide `init` to work with empty collections.
+Apply function `f` to each element(s) in `itrs`, and then repeatedly call the 2 argument
+function `op` with those results or results from previous `op` evaluations until a single value is returned.
+
+If provided, `init` is the return value for empty collections and is used one or more times as
+an argument to `op` for non-empty collections. The `init` value is not transformed by the function `f`.
+Using `init` ensures that all calls to `op` have one argument that is either `init`
+or the result from a previous `op` evaluation, and the ordering of these arguments is
+unspecified. As it may appear in the reduction one or more times, it must be a neutral element for `op` that
```

Comment:
```suggestion
unspecified. As it may appear in the reduction one or more times, it must be a neutral (identity) element for `op` that
```
I think [identity element](https://en.wikipedia.org/wiki/Identity_element) is the more common name in math?

Reply:
```suggestion
unspecified. As it may appear in the reduction one or more times, it must be a neutral element (identity element) for `op` that
```

to improve `grep`ability?

Reply:
I would rather just go all-in on identity — it is indeed a far more common term.  That's what I've done in my latest push.